evalml/pipelines/time_series_regression_pipeline.py

```diff
@@ -0,0 +1,110 @@
+import pandas as pd
+
+from evalml.objectives import get_objective
+from evalml.pipelines.regression_pipeline import RegressionPipeline
+from evalml.problem_types import ProblemTypes
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper,
+    any_values_are_nan,
+    keep_non_nan_rows,
+    pad_with_nans
+)
+
+
+class TimeSeriesRegressionPipeline(RegressionPipeline):
+    """Pipeline base class for time series regression problems."""
+
+    problem_type = ProblemTypes.TIME_SERIES_REGRESSION
+
+    def __init__(self, parameters, random_state=0):
+        """Machine learning pipeline for time series regression problems made out of transformers and a classifier.
+
+        Required Class Variables:
+            component_graph (list): List of components in order. Accepts strings or ComponentBase subclasses in the list
+
+        Arguments:
+            parameters (dict): Dictionary with component names as keys and dictionary of that component's parameters as values.
+                 An empty dictionary {} implies using all default values for component parameters. Pipeline-level
+                 parameters such as gap and max_delay must be specified as a dictionary with the "pipeline" key.
+            random_state (int, np.random.RandomState): The random seed/state. Defaults to 0.
+        """
+        if "pipeline" not in parameters:
+            raise ValueError("gap and max_delay parameters cannot be omitted from the parameters dict. "
+                             "Please specify them as a dictionary with the key 'pipeline'.")
+        pipeline_params = parameters.pop("pipeline")
+        self.gap = pipeline_params['gap']
+        self.max_delay = pipeline_params['max_delay']
+        super().__init__(parameters, random_state)
```

Comment:
If we do `parameters.pop` as above, the pipeline-level params won't be saved in the `parameters` attribute of the pipeline instance. Can we make it so those parameters are saved just like any other component's parameters?

Reply:
Agreed; we should add a test to make sure `gap` and `max_delay` are in the parameters dict after initialization to check for this

Reply:
Great point! I had to modify PipelineBase a little to get this to work (the pipeline parameters method would only read parameters from the component graph) but I'll push this up momentarily!

Reply:
> I decided to only save the pipeline-level parameters if they are present to avoid introducing breaking changes.

I.e. whatever the user passes in is what's saved on the pipeline? If so, yep I agree. We shouldn't create an empty "pipelines" entry in parameters dicts, only create if its populated

Reply:
Yes that's what I meant!